### PR TITLE
Implements `debug` (and the appends the `response` to error `data` field)

### PR DESCRIPTION
**Pull request by sequencemedialimited**
- Ensures that the response status code _etc_ is returned to the Runner which can decide whether to handle the error or throw

### CODE REVIEW

File: lib/user/jwt/client.js [hunk 1]
```diff
@@ -597,30 +607,30 @@ class Client {
     } = e
 
     if (statusCode) {
+      log(statusCode)
+
       if (statusCode > 400 && statusCode < 500) {
         // Data does not exist - ie. expired
-        this.throwRequestError(statusCode)
+        this.throwRequestError(statusCode, statusCode, response)
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: No `message` so provide `statusCode` as the second argument

File: lib/user/jwt/client.js [hunk 2]
```diff
@@ -597,30 +607,30 @@ class Client {
...
       } else {
         if (e.error) {
           // Handle errors which have an error object
           const message = e.error.name || e.error.code || 'EUNSPECIFIED'
-          this.throwRequestError(statusCode, message)
+          this.throwRequestError(statusCode, message, response)
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: Pass the response object down

File: lib/user/jwt/client.js [hunk 3]
```diff
@@ -597,30 +607,30 @@ class Client {
...
         } else {
           // Handle errors which have no error object
           const message = e.code || 'ENOERROR'
-          this.throwRequestError(statusCode, message)
+          this.throwRequestError(statusCode, message, response)
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: Pass the response object down

File: lib/user/jwt/client.js [hunk 4]
```diff
@@ -597,30 +607,30 @@ class Client {
...
         }
       }
     } else if (e.error) {
       // Handle errors which have an error object
       const message = e.error.name || e.error.code || 'EUNSPECIFIED'
-      const statusCode = getErrorStatusCode(message)
-      this.throwRequestError(statusCode, message)
+      this.throwRequestError(getErrorStatusCode(message), message, response)
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: There's no point declaring this as a variable, we may as well just pass the return from the function into the next function, and pass the response object down, too

File: lib/user/jwt/client.js [hunk 5]
```diff
@@ -597,30 +607,30 @@ class Client {
...
     } else {
       // Handle errors which have no error object
       const message = e.code || 'ENOERROR'
-      const statusCode = getErrorStatusCode(message)
-      this.throwRequestError(statusCode, message)
+      this.throwRequestError(getErrorStatusCode(message), message, response)
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: Ditto

File: lib/user/jwt/client.js [hunk 6]
```diff
@@ -637,13 +647,15 @@ class Client {
    * Returns nothing as it should throw an error
    *
    **/
-  throwRequestError (code, message) {
-    message = message || code
+  throwRequestError (code, message = code, data) {
+    error({ code, message })
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: Logs as an object (so, effectively labelling the output)

File: lib/user/jwt/client.js [hunk 7]
```diff
@@ -637,13 +647,15 @@ class Client {
...
+
     throw new this.ErrorClass({
       error: {
         code,
         message
-      }
+      },
+      ...(data ? { data } : {})
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: This will not add the `data` key to the object if the `data` variable is falsy

File: lib/user/jwt/client.js [hunk 8]
```diff
@@ -513,7 +519,11 @@ class Client {
 
       if (logger) logError('API', e)
 
-      client.handleRequestError(e)
+      const { code, message, response } = e
+
+      error({ ...(code ? { code } : {}), ...(message ? { message } : {}) })
```
Comment (sequencemedialimited) [inline lib/user/jwt/client.js]: This will not add the `data` key nor the `message` key to the object if the corresponding variable is falsy

**Comment by sequencemedialimited** [comment]
This package needs to have its patch number incremented before this PR is merged

File: lib/user/jwt/client.js (hunk 8 above)
Comment (brenetic) [reply lib/user/jwt/client.js]: What does the `...` represent in JS?

**Pull request merged**